source/common/config/grpc_subscription_impl.cc

```diff
@@ -121,5 +126,21 @@ void GrpcSubscriptionImpl::disableInitFetchTimeoutTimer() {
   }
 }
 
+GrpcCollectionSubscriptionImpl::GrpcCollectionSubscriptionImpl(
+    const xds::core::v3::ResourceLocator& collection_locator, GrpcMuxSharedPtr grpc_mux,
+    SubscriptionCallbacks& callbacks, OpaqueResourceDecoder& resource_decoder,
+    SubscriptionStats stats, Event::Dispatcher& dispatcher,
+    std::chrono::milliseconds init_fetch_timeout, bool is_aggregated)
+    : GrpcSubscriptionImpl(
+          grpc_mux, callbacks, resource_decoder, stats,
+          TypeUtil::descriptorFullNameToTypeUrl(collection_locator.resource_type()), dispatcher,
+          init_fetch_timeout, is_aggregated, false),
+      collection_locator_(collection_locator) {}
+
+void GrpcCollectionSubscriptionImpl::start(const std::set<std::string>& resource_names) {
+  ASSERT(resource_names.empty());
```

Comment:
This ASSERT seems really strange. I assume that it is related to this implementing still being a work-in-progress.

Reply:
It reflects that xDS collections are an evolution from the existing wildcard subscriptions; there is no way to specify which resources you are subscribing to with LDS/CDS, you subscribe to `*`. This is done through empty resource names.